subsys/bluetooth/controller/util/memq.c

```diff
@@ -58,6 +58,9 @@ void *memq_dequeue(void *tail, void **head, void **mem)
 
 	/* use memq peek to get the link and mem */
 	link = memq_peek(tail, *head, mem);
+	if (!link) {
+		return 0;
```

Comment:
We prefer NULL over 0 for pointers, so that it's clear what the type of the value is (pointer vs integer).